### ISSUE TITLE
[DA] Planetscale engine tests: multi schema

### ISSUE THREAD

**Issue by miguelff** [prisma/prisma-engines#4470]
This is a subcluster of failures after having enabled planetscale tests in https://github.com/prisma/prisma-engines/pull/4423
 
 
Failing tests in this cluster:
 
```
new::multi_schema::multi_schema::create_and_get_many_relations
new::multi_schema::multi_schema::create_and_get_many_to_many_relations
new::multi_schema::multi_schema::crud_many_simple
new::multi_schema::multi_schema::crud_relations
new::multi_schema::multi_schema::crud_simple
new::multi_schema::multi_schema::implicit_m2m_simple
new::multi_schema::multi_schema::test_filter_in
```

Follow the process in [This notion scratchpad page](https://www.notion.so/prismaio/Process-for-tackling-Planetscale-failing-tests-75e280e923e64e2aadf9e367a42aef8b?pvs=4)

**Comment by jkomyno** [comment]
The `multiSchema` preview feature explicitly panics when used with MySQL-adjacent providers, by design, as it never got implemented.

Hence, PlanetScale is excluded from these tests because of a limitation of our implementation on its underlying Quaint+PSL connector.

---

You may have noticed that the header of the `multi_schema` tests is as follows:

```rust
// "multiSchema migrations and introspection are not implemented on MySQL yet"
#[test_suite(
    capabilities(MultiSchema),
    exclude(Mysql, Vitess("planetscale.js", "planetscale.js.wasm"))
)]
```

For a more appropriate clarity to developers touching these tests, we'd only need to declare that the `MultiSchema` capability is required, so that we don't need to manually exclude the providers/flavors that wouldn't succeed on test cases related to this preview feature:

```diff
- // "multiSchema migrations and introspection are not implemented on MySQL yet"
#[test_suite(
  capabilities(MultiSchema),
-   exclude(Mysql, Vitess("planetscale.js", "planetscale.js.wasm"))
)]
```

Unfortunately, this will only be possible once we understand the impact of removing these lines:

https://github.com/prisma/prisma-engines/blob/ca42e86a7b1bc9a4d32728da72fc42feaa05f011/psl/psl-core/src/builtin_connectors/mysql_datamodel_connector.rs#L53-L55

**Comment by janpio** [comment]
For our users PlanetScale is just MySQL. So if we support `multiSchema` for MySQL, they also expect it to work on a database hosted at PlanetScale.

**Comment by jkomyno** [comment]
> For our users PlanetScale is just MySQL. So if we support multiSchema for MySQL, they also expect it to work on a database hosted at PlanetScale.

I'd agree with you in theory, but we do NOT support `multiSchema` on MySQL. Please see: https://github.com/prisma/prisma/issues/16943.

In other words, it's totally normal for PlanetScale to fail on `multiSchema` tests, and that's by design, because the MySQL connector itself fails on them even when using the Rust driver.

**Comment by janpio** [comment]
Why do we run the tests for it then? We do not have equivalent failing tests on the Rust side.

**Comment by jkomyno** [comment]
> Why do we run the tests for it then? We do not have equivalent failing tests on the Rust side. 

We have equivalent tests on the Rust side which would fail, were they toggled off (or, so to speak, "commented out").

With [this message above](https://github.com/prisma/prisma-engines/issues/4470#issuecomment-1971181589) in mind, we see that the `multi_schema` tests start with:

```rust
#[test_suite(
    capabilities(MultiSchema),
    exclude(Mysql, Vitess("planetscale.js", "planetscale.js.wasm"))
)]
```

In plain English, this translates to:
- Run each test case wrapped by this `test_suite` macro only in those connectors that support the `MultiSchema` capability, i.e., that are able to perform multi-schema queries without crashing.
- From this set of connectors, we exclude `Mysql` (i.e., MySQL / MariaDB databases using a Rust driver), and `planetscale.js*` (Napi/Wasm driver adapters for PlanetScale/Vitess databases).

A reasonable question/doubt would be:
> Why do we need to exclude these connectors? Didn't we just say that Prisma MySQL-adjacent connectors do NOT support the `multiSchema` preview feature?

That's correct. However, for some reason, MySQL [is marked with the `MultiSchema` capability](https://github.com/prisma/prisma-engines/blob/ca42e86a7b1bc9a4d32728da72fc42feaa05f011/psl/psl-core/src/builtin_connectors/mysql_datamodel_connector.rs#L53-L55), which imho is a mistake.

The immediate choice, in my opinion, is to get rid of [this line](https://github.com/prisma/prisma-engines/blob/ca42e86a7b1bc9a4d32728da72fc42feaa05f011/psl/psl-core/src/builtin_connectors/mysql_datamodel_connector.rs#L53-L55), so that we can transform the test header from

```rust
// "multiSchema migrations and introspection are not implemented on MySQL yet"
#[test_suite(
    capabilities(MultiSchema),
    exclude(Mysql, Vitess("planetscale.js", "planetscale.js.wasm"))
)]
```

to

```diff
- // "multiSchema migrations and introspection are not implemented on MySQL yet"
#[test_suite(
  capabilities(MultiSchema),
-   exclude(Mysql, Vitess("planetscale.js", "planetscale.js.wasm"))
)]
```
